OpenSim/Simulation/Model/Geometry.h

```diff
@@ -180,8 +150,34 @@ class OSIMSIMULATION_API Geometry : public Component {
     /// Methods to support frame as a connection, implement Component interface
     void constructConnectors() override;
 
+protected:
+    /// Map this Geometry into a list of primitives aka SimTK::DecorativeGeometry 
+    /// and return the result in the passed in Array.
+    virtual void createDecorativeGeometry(
+        SimTK::Array_<SimTK::DecorativeGeometry>&) const = 0;
```

Comment:
@aseth1 We have off-the-shelf Geometry that work based on xml properties and fully customized Geometry by using generateDecorations mechanism. Having a hybrid maybe useful down the line at some point but it'd be following the custom route. For off-the-shelf Geometry it's clean that what you put in XML is what you get without any meddling. Also makes the interface simpler IMHO.